src/Couchbase.Lite.Shared/API/Sync/ReplicatorProgress.cs

```diff
@@ -70,28 +71,37 @@ public struct ReplicatedDocument
         [CanBeNull]
         public CouchbaseException Error { get; }
 
+        internal Exception Exception { get; set; }
```

Comment:
How does this help the user see the original message?  There needs to be a way to preserve the original exception in a way that is visible to the user (e.g. `Error`'s `InnerException`)